src/otx/core/model/entity/base.py

```diff
@@ -263,6 +280,33 @@ def _reset_prediction_layer(self, num_classes: int) -> None:
     def _optimization_config(self) -> dict[str, str]:
         return {}
 
+    def _reset_model_forward(self) -> None:
+        forward_with_auxiliaries = self._get_forward_with_auxiliaries()
+        if forward_with_auxiliaries is None:
+            msg = "forward_with_auxiliaries needs to be implemented for dumping auxiliaries."
+            raise NotImplementedError(msg)
+
+        self.original_model_forward = self.model.forward
+
+        func_type = types.MethodType
+        self.model.forward = func_type(forward_with_auxiliaries, self.model)
+
+    def _restore_model_forward(self) -> None:
+        if not self.original_model_forward:
+            msg = "original_model_forward is None."
+            raise RuntimeError(msg)
+
+        func_type = types.MethodType
+        self.model.forward = func_type(self.original_model_forward, self.model)
+        self.original_model_forward = None
+
+    @staticmethod
+    def _update_onnx_output_names(onnx_export_configuration: dict) -> None:
+        if "output_names" not in onnx_export_configuration:
```

Comment:
This implementation is suitable only for classification models, so it should be moved to the corresponding class.

Reply:
Right, fixed